src/meta/meta_turtle.c

```diff
@@ -226,6 +258,13 @@ __wt_turtle_init(WT_SESSION_IMPL *session)
         if (ret != 0) {
             WT_RET(__wt_remove_if_exists(session, WT_METADATA_TURTLE, false));
             loadTurtle = true;
+        } else {
+            /*
+             * Check the version in the turtle file to validate whether we can start up on the
+             * turtle file version seen. Return an error if we can't. Only check if we either didn't
+             * run salvage or if salvage didn't fail to read it.
+             */
+            WT_RET(__turtle_validate_version(session));
```

Comment:
I think that this code should go below the exist checks right after here. There are cases where if this is a backup then we remove the turtle file. So maybe at this location based, we set a variable `validate` to true and then later call this function if that variable is set.

Reply:
I moved the check to a different location let me know if that was what you had in mind.